### PR TITLE
feat: 🎸 Add host type basic support

### PR DESCRIPTION
**Pull request by calcaide**
Add host type basic support.

### CODE REVIEW

File: addons/api/mirage/factories/host.js [hunk 1]
```diff
@@ -10,4 +10,9 @@ export default factory.extend({
       'delete',
     ],
   id: (i) => `host-id-${i}`,
+  plugin() {
+    if (this.type === 'plugin') {
+      return this.hostCatalog.plugin;
```
Comment (DhariniJeeva) [inline addons/api/mirage/factories/host.js:15]: just curious, why are we returning `hostCatalog.plugin` instead of `host.plugin`

Comment (calcaide) [reply addons/api/mirage/factories/host.js:15]: Because we want the object `plugin` inherit from hostCatalog.

Comment (randallmorey) [reply addons/api/mirage/factories/host.js:15]: @DhariniJeeva the type is inherited from the host catalog.  However this raises a problem with the implementation.  We need to apply this inheritance logic to `type` as well @calcaide.

Comment (calcaide) [reply addons/api/mirage/factories/host.js:15]: @randallmorey 
On it's way! I did it in `host-set` [here](https://github.com/hashicorp/boundary-ui/pull/925/files#diff-c4d10025ef7bd2a9d9275ee8670da51e3d034c096ac9fe931dc4233ea88544b0R8) I will reproduce that pattern here.

**Comment by DhariniJeeva** [comment]
Looks good to me! one minor query- does the host set need support for `unknown` type as well?

**Comment by randallmorey** [comment]
> Looks good to me! one minor query- does the host set need support for `unknown` type as well?

I'm beginning to think we do.  Taking another look.

**Comment by randallmorey** [comment]
> Looks good to me! one minor query- does the host set need support for `unknown` type as well?

For now, no.

**Review by randallmorey — APPROVED** [review]

**Pull request merged**